src/lib/logical_query_plan/join_node.cpp

```diff
@@ -123,6 +123,44 @@ std::shared_ptr<LQPUniqueConstraints> JoinNode::unique_constraints() const {
   return unique_constraints;
 }
 
+std::vector<FunctionalDependency> JoinNode::non_trivial_functional_dependencies() const {
+  if (join_mode != JoinMode::Semi && join_mode != JoinMode::AntiNullAsTrue && join_mode != JoinMode::AntiNullAsFalse) {
+    /**
+     * In the course of inner/outer joins,
+     *  (a) some unique constraints and therefore some trivial FDs of the JoinNode's input nodes might become lost.
+     *  (b) some unique constraints might stay valid. Due to the increased number of output expressions, however,
+     *      derived FDs become longer. For example: - Table A: (UNIQUE(a), b) leads to the FD {a} -> {b}
+     *                                              - Table B: (x, y)
+     *                                              After joining both tables, we may end up with
+     *                                              - (UNIQUE(a), b, x, y) which leads to the FD {a} -> {b, x, y}
```

Comment:
Isn't this only the case if x is unique too? Also, is this something that we already do? I can't find it in the code.

Reply:
Clarify that this is not yet done